dataplane/ebpf/src/egress/icmp.rs

```diff
@@ -8,42 +8,43 @@ use core::mem;
 
 use aya_bpf::{bindings::TC_ACT_PIPE, helpers::bpf_csum_diff, programs::TcContext};
 use aya_log_ebpf::info;
+use network_types::{ip::Ipv4Hdr, eth::EthHdr, icmp::IcmpHdr};
 
 use crate::{
-    bindings::{icmphdr, iphdr},
-    utils::{csum_fold_helper, ptr_at, ETH_HDR_LEN, IP_HDR_LEN},
+   
+    utils::{csum_fold_helper, ptr_at, IP_HDR_LEN},
     BLIXT_CONNTRACK,
 };
 
-const ICMP_HDR_LEN: usize = mem::size_of::<icmphdr>();
+
 const ICMP_PROTO_TYPE_UNREACH: u8 = 3;
 
 pub fn handle_icmp_egress(ctx: TcContext) -> Result<i32, i64> {
-    let ip_hdr: *mut iphdr = unsafe { ptr_at(&ctx, ETH_HDR_LEN) }?;
-
-    let icmp_header_offset = ETH_HDR_LEN + IP_HDR_LEN;
```

Comment:
I've noticed throughout this PR that these "offset" variables are being removed in favor of inline addition. This is a somewhat minor thing but I think it subtly reduces the readability of the code. For some of us who write a lot of network code seeing the eth header length and ipv4 header length added together is intuitive, but for some having a clear named variable to describe mnemonically the meaning of the value can be a small, but helpful thing. My suggestion would be to put these variables back:

```suggestion
    let icmp_header_offset = EthHdr::LEN + Ipv4Hdr::LEN;
```

But let me know your thoughts?

Reply:
makes sense. Updated